### ISSUE TITLE
Optionals not appending following items

### ISSUE THREAD

**Issue by gcrevell** [joeldev/JLRoutes#94]
When attempting to register the route `”/(rest/)(app/):object/:id”`, it does not produce the routes I would expect which would be

```
"<JLRRouteDefinition 0x600000650c80> - /rest/app/:object/:id (priority: 0)",
"<JLRRouteDefinition 0x60000064e130> - /app/:object/:id (priority: 0)",
"<JLRRouteDefinition 0x60000064ebb0> - /rest/:object/:id (priority: 0)",
"<JLRRouteDefinition 0x6000006501d0> - /:object/:id (priority: 0)"
```

Instead, I am getting routes like 

```
"<JLRRouteDefinition 0x600000650c80> - /rest/app/ (priority: 0)",
"<JLRRouteDefinition 0x60000064e130> - /app/ (priority: 0)",
"<JLRRouteDefinition 0x60000064ebb0> - /rest/ (priority: 0)",
"<JLRRouteDefinition 0x6000006501d0> - / (priority: 0)"
```

I tried a few different combinations with moving the `/` around. 

```
“(/rest)(/app)/:object/:id”
“/(rest)(app)/:object/:id”       // I don’t care about the combination of rest and app, so this combination is fine
```

But they all produce similar results.

Is this expected? I can obviously just register each route separately but would prefer to condense the code if possible.

BTW, running JLRoutes 2.0.5

**Comment by joeldev** [comment]
Sorry for the long response delay. Will take a look! Thanks for reporting it.

**Comment by joeldev** [comment]
Wow yeah, this is a massive bug. Thanks for reporting it! I am currently working on rewriting my optional parsing and route expansion code in response to this. I've added this as a new test case and will add more to cover some of these other situations that the old code did not handle.

**Comment by gcrevell** [comment]
Hey thanks for getting back to me! I'm looking forward to a fix. I worked around this by just adding each route manually which worked fine but is not exactly desirable.

**Comment by joeldev** [comment]
This fix is now available in pod release 2.0.6.
